### PR TITLE
OP#42726 manual testplan

### PR DESCRIPTION
**Pull request by individual-it**
manual testplan for the OpenProject integration in NextCloud

fixes https://community.openproject.org/projects/nextcloud-integration/work_packages/42726

### CODE REVIEW

File: testplan.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
+## Installation
+
+### Prerequisites
+
+- fresh NextCloud, without any previous history of having the app installed
+
```
Comment (phil-davis) [inline testplan.md:6]:
```suggestion
- log in as admin and/or be at the command prompt

```

File: testplan.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+| disable the app using the WebUI                        | app is listed as disabled            |        |         |
+
+## Configuration
+### Before configuring the app
+
+#### Prerequisites
+
+- fresh NextCloud, without any previous history of having the app installed
+- install, but don't configure OpenProject integration app
+
```
Comment (phil-davis) [inline testplan.md:25]:
```suggestion
- log in as an admin

```

File: testplan.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+| check administrators settings       | OpenProject section shown, only OpenProject host can be entered             |        |         |
+| check personal settings             | OpenProject section in 'Connected accounts' shown, good error message shown |        |         |
+
+### Configuration as admin
+#### Prerequisites
+
+- NextCloud installed
+- OpenProject integration app installed
+- OpenProject integration app enabled
+
```
Comment (phil-davis) [inline testplan.md:41]:
```suggestion
- log in as an admin

```

File: testplan.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+| enter also client secret                                                         | settings can be saved, NextCloud OAuth settings are generated and shown |        |         |
+
+### Connecting as user
+#### Prerequisites
+
+- NextCloud installed
+- OpenProject integration app installed
+- OpenProject integration app enabled
+- OpenProject integration app configured as administrator
+
```
Comment (phil-davis) [inline testplan.md:59]:
```suggestion
- log in as a regular user

```

File: testplan.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+### Edit configuration as admin
+#### Prerequisites
+
+- NextCloud installed
+- OpenProject integration app installed
+- OpenProject integration app enabled
+- OpenProject integration app configured as administrator
+- multiple NC users connected to OpenProject
+
```
Comment (phil-davis) [inline testplan.md:82]:
```suggestion
- log in as an admin

```

File: testplan.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+## Usage
+
+### Prerequisites
+
+- NextCloud installed
+- OpenProject integration app installed
+- OpenProject integration app enabled
+- OpenProject integration app configured as administrator
+- multiple NC users connected to OpenProject
+
```
Comment (phil-davis) [inline testplan.md:100]:
```suggestion
- log in as a regular user

```

File: testplan.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+| link multiple WP to a file, delete some WP                                       | WP listed correctly                              |        |         |
+| link multiple WP to a file, search for similar WP like the listed ones           | connected WP are not listed in the search        |        |         |
+
+### Notifications
+
+### Prerequisites
+
+- notifications app installed and enabled https://github.com/nextcloud/notifications
+- enable notifications in "OpenProject integration" section of "Connected accounts"
+
```
Comment (phil-davis) [inline testplan.md:132]:
```suggestion
- log in as a regular user

```

**Review by phil-davis — COMMENTED** [review]
Suggestion: specify somewhere what sort of user to be for each section of tests.

Also, we could test all the "admin" things with a user that is a member of the "admin" group, but is not the literal user "admin", just to be sure that none of the settings are hard-coded in some way to the user called "admin".

**Comment by individual-it** [comment]
@phil-davis I've added a new section to every test. does that make it clearer?

**Review by phil-davis — APPROVED** [review]
LGTM

**Pull request merged**